src/charm.py

```diff
@@ -167,6 +169,7 @@ def __init__(self, *args: Any):
         self.framework.observe(
             self.on.identity_platform_admin_ui_relation_changed, self._on_peer_relation_changed
         )
+        self.framework.observe(self.on.collect_unit_status, self._on_collect_status)
```

Comment:
question: don't we need to also observe `collect_app_status`? Or do we just let the unit's status and let juju decide the app's status based on that?

Reply:
yep, collect unit status will be good enough I think.